tool/CMakeLists.txt

```diff
@@ -29,17 +31,62 @@ endif()
 
 if(APPLE OR WIN32 OR ANDROID)
   target_link_libraries(bssl ssl crypto)
+  set(LIBRT_FLAGS "")
 else()
   find_library(FOUND_LIBRT rt)
   if(FOUND_LIBRT)
     target_link_libraries(bssl ssl crypto -lrt)
+    set(LIBRT_FLAGS "-lrt")
   else()
     target_link_libraries(bssl ssl crypto)
+    set(LIBRT_FLAGS "")
   endif()
 endif()
 
-install(TARGETS bssl
-        EXPORT bssl-targets
-        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-)
+function(build_benchmark target_name additional_include_dir lib_crypto)
+  message(${additional_include_dir})
+  add_executable(
+          ${target_name}
+          speed.cc
+          args.cc
+          const.cc
+          benchmark.cc
+  )
+  # Link with the internal tool directory for shared headers with the rest of the tool instead of the top level AWS-LC
+  # include directory
+  target_include_directories(${target_name} PUBLIC ${additional_include_dir} ${AWSLC_INSTALL_DIR}/include/internal/tool)
+  target_link_libraries(${target_name} ${lib_crypto} ${LIBRT_FLAGS})
+
+  string(TOUPPER ${target_name} upper_target_name)
+  target_compile_options(${target_name} PUBLIC -D${upper_target_name})
+endfunction()
+
+if(AWSLC_INSTALL_DIR)
+  build_benchmark(awslc_bm ${AWSLC_INSTALL_DIR}/include crypto)
+endif()
+
+# This expects a directory which contains lib/libcrypto.a and the includes in include/openssl/*.h
+if(OPENSSL_INSTALL_DIR)
+  build_benchmark(ossl_bm ${OPENSSL_INSTALL_DIR}/include ${OPENSSL_INSTALL_DIR}/build/lib/libcrypto.a)
+
+  # libpthread is a linux only library
+  if(NOT MSVC AND NOT ANDROID)
+    # OpenSSL uses pthread_rwlock_unlock from pthread and  dlopen/close from dl
+    target_link_libraries(ossl_bm -DOPENSSL_BENCHMARK pthread dl)
+    # OpenSSL uses both anonymous structs and unions
+    target_compile_options(ossl_bm PUBLIC -DOPENSSL_BENCHMARK -Wno-gnu-anonymous-struct -Wno-nested-anon-types)
+  endif()
+endif()
+
+if(BORINGSSL_INSTALL_DIR)
+  build_benchmark(bssl_bm ${BORINGSSL_INSTALL_DIR}/include ${BORINGSSL_INSTALL_DIR}/build/crypto/libcrypto.a)
+
+  # libpthread is a linux only library
+  if(NOT MSVC AND NOT ANDROID)
+    # BoringSSL uses pthread_rwlock_unlock from pthread
+    target_link_libraries(bssl_bm pthread)
+    # BoringSSL uses both anonymous structs and unions
+    target_compile_options(bssl_bm PUBLIC -Wno-gnu-anonymous-struct -Wno-nested-anon-types)
```

Comment:
The speed tool uses anom structs and unions? Not sure I understand why these are needed.